### PR TITLE
hv: clean up pci

### PR DESCRIPTION
**Pull request by shiqingg**
- add one more filed in "union pci_bdf"
- remove following interfaces:
      * pci_bus
      * pci_slot
      * pci_func
      * pci_devfn

 - update the argument type to union for function "bdf_is_equal"
      Declaring argument as pointer is not necessary since it
      only does the comparison.

Tracked-On: #1842
Signed-off-by: Shiqing Gao <shiqing.gao@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
28 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><2> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><3> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><4> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><5> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><6> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><7> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><8> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><9> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Cast to an unrelated type. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><10> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Cast to an unrelated type. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><11> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation to a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><12> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation to a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><13> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation on a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><14> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation on a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><15> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:872<br> reason:Function return value potentially unused. : dmar_bdf.fields<br><16> config:part_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:872<br> reason:Function return value potentially unused. : dmar_bdf.value<br><17> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:872<br> reason:Function return value potentially unused. : dmar_bdf.value<br><18> config:part_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:872<br> reason:Function return value potentially unused. : dmar_bdf.fields<br><19> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_free_irte offset:83<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><20> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_free_irte offset:83<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><21> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_msi offset:105<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><22> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_msi offset:105<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><23> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_rte offset:177<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><24> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_rte offset:177<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><25> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_msix_remapping offset:330<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><26> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_msix_remapping offset:330<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><27> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_intx_remapping offset:426<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><28> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_intx_remapping offset:426<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/arch/x86/vtd.c [hunk 1]
```diff
@@ -1055,13 +1058,17 @@ static int32_t add_iommu_device(struct iommu_domain *domain, uint16_t segment, u
 	uint64_t hi_64;
 	uint64_t lo_64 = 0UL;
 	int32_t ret = 0;
+	union pci_bdf sid;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/vtd.c]: what does sid mean? add a comment?

Comment (shiqingg) [reply hypervisor/arch/x86/vtd.c]: > what does sid mean? add a comment?

It is source id. Will add a comment.

Comment (shiqingg) [reply hypervisor/arch/x86/vtd.c]: @dongyaozu A new version is pushed with the comment added. Could you please help review again?

**Comment by HuiHuangShi** [comment]
28 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><2> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><3> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><4> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><5> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><6> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><7> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><8> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><9> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Cast to an unrelated type. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><10> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Cast to an unrelated type. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><11> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation to a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><12> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation to a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><13> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation on a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><14> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation on a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><15> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:872<br> reason:Function return value potentially unused. : dmar_bdf.fields<br><16> config:part_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:872<br> reason:Function return value potentially unused. : dmar_bdf.value<br><17> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:872<br> reason:Function return value potentially unused. : dmar_bdf.value<br><18> config:part_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:872<br> reason:Function return value potentially unused. : dmar_bdf.fields<br><19> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_free_irte offset:83<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><20> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_free_irte offset:83<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><21> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_msi offset:105<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><22> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_msi offset:105<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><23> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_rte offset:177<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><24> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_rte offset:177<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><25> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_msix_remapping offset:330<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><26> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_msix_remapping offset:330<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><27> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_intx_remapping offset:426<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><28> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_intx_remapping offset:426<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
28 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_free_irte offset:83<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><2> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_free_irte offset:83<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><3> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_msi offset:105<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><4> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_msi offset:105<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><5> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_rte offset:177<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><6> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_rte offset:177<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><7> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_msix_remapping offset:330<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><8> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_msix_remapping offset:330<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><9> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_intx_remapping offset:426<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><10> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_intx_remapping offset:426<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><11> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Cast to an unrelated type. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><12> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Cast to an unrelated type. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><13> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation to a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><14> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation to a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><15> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation on a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><16> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation on a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><17> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:873<br> reason:Function return value potentially unused. : dmar_bdf.value<br><18> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:873<br> reason:Function return value potentially unused. : dmar_bdf.fields<br><19> config:part_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:873<br> reason:Function return value potentially unused. : dmar_bdf.fields<br><20> config:part_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:873<br> reason:Function return value potentially unused. : dmar_bdf.value<br><21> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><22> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><23> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><24> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><25> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><26> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><27> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><28> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
28 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_free_irte offset:83<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><2> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_free_irte offset:83<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><3> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_msi offset:105<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><4> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_msi offset:105<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><5> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_rte offset:177<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><6> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:ptirq_build_physical_rte offset:177<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><7> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_msix_remapping offset:330<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><8> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_msix_remapping offset:330<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><9> config:sharing_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_intx_remapping offset:426<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><10> config:part_config filename:/hypervisor/arch/x86/guest/assign.c func:remove_intx_remapping offset:426<br> reason:Procedure contains UR data flow anomalies. : intr_src.src.msi.fields.devfun<br><11> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Cast to an unrelated type. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><12> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Cast to an unrelated type. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><13> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation to a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><14> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation to a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><15> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation on a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & bdf<br><16> config:common filename:/hypervisor/debug/shell.c func:get_ptdev_info offset:1047<br> reason:Casting operation on a pointer. : (struct pci_bdf* to unsigned int*): ( uint32_t * ) & vbdf<br><17> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:873<br> reason:Function return value potentially unused. : dmar_bdf.value<br><18> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:873<br> reason:Function return value potentially unused. : dmar_bdf.fields<br><19> config:part_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:873<br> reason:Function return value potentially unused. : dmar_bdf.fields<br><20> config:part_config filename:/hypervisor/arch/x86/vtd.c func:fault_record_analysis offset:873<br> reason:Function return value potentially unused. : dmar_bdf.value<br><21> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><22> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><23> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><24> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_assign_ptdev offset:870<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><25> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><26> config:sharing_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><27> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.devfun<br><28> config:part_config filename:/hypervisor/common/hypercall.c func:hcall_deassign_ptdev offset:912<br> reason:Procedure contains UR data flow anomalies. : bdf.fields.bus<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by acrnsi** [comment]
OK to verify

**Pull request merged**